### PR TITLE
chore(dashboard): adjust prebuild table col width

### PR DESCRIPTION
**Pull request by andreafalzetti**
## Description
The Prebuilds page not always align correctly the columns, e.g. see screeenshot below:

Screenshot 2022-06-09 at 11 21 36 copy

I think this small CSS change could improve already some scenarios.

## Related Issue(s)

N/A

## How to test
1. Start the dashboard for gitpod.io

(I would also suggest Prev Env, but there are no prebuilds there)

| Before | After |
|--------|-------|
|     Screenshot 2022-06-09 at 11 21 36   |   Screenshot 2022-06-09 at 11 21 03    |

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by geropl** [comment]
@AlexTugarev Could you review this PR (tomorrow)? :pray:

**Comment by jankeromnes** [comment]
Happy to take a look 👀

File: components/dashboard/src/projects/Prebuilds.tsx
Comment (AlexTugarev) [inline components/dashboard/src/projects/Prebuilds.tsx:193]: It would be great to not overshoot at `md`.

**Comment by AlexTugarev** [comment]
@jankeromnes, are you still looking at this? This seems to fix a bigger issue, and I'm keen to merge it as is. wdyt?

**Comment by jankeromnes** [comment]
Ah, yes, thanks for the reminder. Triggered a build then forgot about this.

Resuming review now. 👀

File: components/dashboard/src/projects/Prebuilds.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/Prebuilds.tsx:193]: Tip for the future: Using CSS grid is much more powerful and robust in these situation than trying to guess good fixed widths.

With grid, you can simply specify "I want three columns of exactly the same width", or specify custom relative column widths, but they'll almost always be correct regardless of screen size, margins, size of content, etc.

**Review by jankeromnes — APPROVED** [review]
Thanks a lot for improving our Prebuilds list! ✨

I tested this change at various resolutions, and with various commit messages / branch name lengths. I think the current implementation is still not perfect (e.g. might need an ellipsis + tooltip on the branch name) hence the hold, but please feel free to merge this when you like (this is easy to adjust later as well).

FYI, here are my test results:

| My normal screen size | Max screen size | Smartphone (>50% of web users) | iPad (portrait) |
| --- | --- | --- | --- |
| Screenshot 2022-06-14 at 11 25 40 | Screenshot 2022-06-14 at 11 26 09 | Screenshot 2022-06-14 at 11 26 43 | Screenshot 2022-06-14 at 11 27 01 |

**Tip**: To automatically add an ellipsis to a potentially super long cell (e.g. the branch name), you can add:
- the `truncate` CSS class
- a `title` attribute with the exact same text content as the element (so that users can still hover over the truncated text to see the full text)

See also how the commit message is truncated for inspiration.

/hold

**Comment by jankeromnes** [comment]
Oh, wait, I wonder if https://github.com/gitpod-io/gitpod/pull/10472 maybe already fixes the same issue? 👀

Looks like we may need a `overflow-hidden` in addition to the `truncate` class. 💡

**Comment by akosyakov** [comment]
should be merged or closed?

**Comment by gtsiolis** [comment]
I'd suggest getting this merged as it's already a good MVC improvement, regardless if the rendering on small (mobile) viewports looks great, leaving this for https://github.com/gitpod-io/gitpod/issues/4050.

If that's useful, I'd like to help get https://github.com/gitpod-io/gitpod/pull/10472 also merged afterwards.

**Comment by akosyakov** [comment]
@jankeromnes @AlexTugarev is it alright to remove hold?

**Comment by jankeromnes** [comment]
Hi! Yes, and sorry for sitting on this for so long.

@gtsiolis's [comment](https://github.com/gitpod-io/gitpod/pull/10564#issuecomment-1167311730) makes a good case or merging this. 👍 Thus, removing the hold.

/unhold

**Pull request merged**
